starknet_devnet/server.py

```diff
@@ -63,7 +63,7 @@ async def add_transaction():
             error_message = err.message
             status = TxStatus.REJECTED
 
-        transaction_hash = await starknet_wrapper.store_invoke_transaction(
+        transaction_hash = await starknet_wrapper.store_wrapper_transaction(
```

Comment:
This should take care of its transaction storing the same way `deploy` does it. Also, `deploy` should accept the original `Transaction` object, then convert it internally to `InternalDeploy`. That way there'd be no need to access `state.general_config` or even `starknet.state` (the get_state method might not even be needed).